pkg/factsModel/facts.go

```diff
@@ -0,0 +1,59 @@
+package factsModel
+
+import (
+	"context"
+	"fmt"
+	"github.com/appuio/appuio-cloud-reporting/pkg/db"
+	"github.com/jmoiron/sqlx"
+)
+
+func GetByFact(ctx context.Context, tx *sqlx.Tx, fact *db.Fact) (*db.Fact, error) {
+	var facts []db.Fact
+	err := sqlx.SelectContext(ctx, tx, &facts,
+		`SELECT facts.* FROM facts WHERE date_time_id = $1 AND query_id = $2 AND tenant_id = $3 AND category_id = $4 AND product_id = $5 AND discount_id = $6`,
+		fact.DateTimeId, fact.QueryId, fact.TenantId, fact.CategoryId, fact.ProductId, fact.DiscountId)
+	if err != nil {
+		return nil, err
+	}
+	if len(facts) == 0 {
+		return nil, nil
+	}
+	return &facts[0], nil
+}
+
+func Ensure(ctx context.Context, tx *sqlx.Tx, ensureFact *db.Fact) (*db.Fact, error) {
+	fact, err := GetByFact(ctx, tx, ensureFact)
+	fmt.Printf(">>> %v\n", fact)
+	if err != nil {
+		fmt.Printf("A\n")
+		return nil, err
+	}
+	if fact == nil {
+		fmt.Printf("B\n")
+		fact, err = Create(tx, ensureFact)
+		if err != nil {
+			return nil, err
+		}
+	}
+	fmt.Printf("C\n")
```

Comment:
What's the purpose of printing `A, B, C`?

Reply:
Removed.